### PR TITLE
Rename messageId into messageCounter

### PR DESCRIPTION
**Pull request by kghost**
#### Problem
`MessageCounter` is the official name in the specs.

#### Change overview
This PR is generated by following commands:

```
git sed -f g messageId messageCounter
git sed -f g MessageId MessageCounter
git sed -f g MsgId MessageCounter
git sed -f g msgId messageCounter
git sed -f g msgID messageCounter
```

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

**Review by Damian-Nordic — CHANGES_REQUESTED** [review]

File: examples/minimal-mdns/client.cpp [hunk 1]
```diff
@@ -48,8 +48,8 @@ struct Options
     mdns::Minimal::QType type = mdns::Minimal::QType::ANY;
 } gOptions;
 
-constexpr uint32_t kTestMessageId   = 0x1234;
```
[the remark this answers is not in the document]
Comment (kghost) [reply examples/minimal-mdns/client.cpp:51]: Thanks, I'll revert this part.

File: src/messaging/ReliableMessageContext.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ CHIP_ERROR ReliableMessageContext::FlushAcks()
         if (err == CHIP_NO_ERROR)
         {
 #if !defined(NDEBUG)
-            ChipLogDetail(ExchangeManager, "Flushed pending ack for MsgId:%08" PRIX32, mPendingPeerAckId);
+            ChipLogDetail(ExchangeManager, "Flushed pending ack for MessageCounter:%08" PRIX32, mPendingPeerAckId);
```
Comment (Damian-Nordic) [inline src/messaging/ReliableMessageContext.cpp:128]: Rename `mPendingPeerAckId`, too?

File: src/messaging/ReliableMessageContext.h [hunk 3]
```diff
@@ -217,9 +217,9 @@ class ReliableMessageContext
 private:
     void RetainContext();
     void ReleaseContext();
-    CHIP_ERROR HandleRcvdAck(uint32_t AckMsgId);
-    CHIP_ERROR HandleNeedsAck(uint32_t messageId, BitFlags<MessageFlagValues> messageFlags);
-    CHIP_ERROR HandleNeedsAckInner(uint32_t messageId, BitFlags<MessageFlagValues> messageFlags);
+    CHIP_ERROR HandleRcvdAck(uint32_t AckMessageCounter);
```
Comment (Damian-Nordic) [inline src/messaging/ReliableMessageContext.h:220]: super nit:
```suggestion
    CHIP_ERROR HandleRcvdAck(uint32_t ackMessageCounter);
```

**Review by Damian-Nordic — APPROVED** [review]

File: src/messaging/ReliableMessageContext.cpp (hunk 2 above)
Comment (kghost) [reply src/messaging/ReliableMessageContext.cpp]: Done

File: src/messaging/ReliableMessageContext.h (hunk 3 above)
Comment (kghost) [reply src/messaging/ReliableMessageContext.h]: Done

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**